### ISSUE TITLE
Implement zero-downtime major upgrades (blue/green)

### ISSUE THREAD

**Issue by vitabaks** [vitabaks/autobase#537]
In the previous release ([1.8.0](https://github.com/vitabaks/postgresql_cluster/releases/tag/1.8.0)), we introduced automation for in place major upgrades with minimal downtime (see [PR #301](https://github.com/vitabaks/postgresql_cluster/pull/301)). \
While this was a significant improvement, there is still a need for a solution that enables zero-downtime upgrades.

**Proposed Enhancement:**
To achieve zero downtime during major PostgreSQL upgrades, we need to implement a feature that utilizes `pg_upgrade` on a standby cluster. 

This approach involves the following steps (in short, without details):

1. Setting up a standby cluster (with pg_basebackup or from backup);
2. Synchronize the standby cluster with the source cluster using physical replication;
3. Create logical replication slot on the source database (and save LSN);
4. Promote the target cluster to LSN from logical replication slot;
5. Perform `pg_upgrade` on the target cluster;
6. Synchronize the target cluster with the source cluster using logical replication (apply changes);
7. Perform a switchover to target cluster with near-zero downtime;
8. Configure reverse logical replication for the possibility of rollback without data loss.

This method will allow us to perform major version upgrades without impacting the availability of the PostgreSQL service. With the ability to quickly rollback without losing the latest changes.

---

_Consider sponsoring the project via [GitHub](https://github.com/sponsors/vitabaks) or [Patreon](https://patreon.com/vitabaks)_

**Comment by vitabaks** [comment]
If you're interested in this feature, please consider becoming a [sponsor](https://postgresql-cluster.org/sponsor).

**Comment by khawarizmus** [comment]
Can this be achieved with supavisor https://github.com/supabase/supavisor without having to use a standby cluster but to gradually upgrade each node?

**Comment by vitabaks** [comment]
> Can this be achieved with supavisor https://github.com/supabase/supavisor without having to use a standby cluster but to gradually upgrade each node?

I don't think so, without using logical replication, a zero-downtime upgrade is hardly possible.

**Comment by vitabaks** [comment]
Done https://github.com/vitabaks/autobase/pull/1430
